WebService/v1/Models/SimulationApiModel/SimulationApiModel.cs

```diff
@@ -268,11 +268,22 @@ private static SimulationStatistics GetSimulationStatistics(Simulation simulatio
         {
             var statistics = SimulationStatistics.FromServiceModel(simulation.Statistics);
 
-            if (statistics != null)
+            if (statistics != null && simulation.ActualStartTime.HasValue)
             {
                 DateTimeOffset now = DateTimeOffset.UtcNow;
-                TimeSpan duration = simulation.IsActiveNow ? now.Subtract(simulation.ActualStartTime.Value) : simulation.StoppedTime.Value.Subtract(simulation.ActualStartTime.Value);
-                statistics.AverageMessagesPerSecond = (double) statistics.TotalMessagesSent / duration.TotalSeconds;
+                var actualStartTime = simulation.ActualStartTime.Value;
+                double durationInSeconds = 0;
+
+                if (simulation.IsActiveNow)
+                {
+                    durationInSeconds = now.Subtract(actualStartTime).TotalSeconds;
+                }
+                else if (simulation.StoppedTime.HasValue)
+                {
+                    durationInSeconds = simulation.StoppedTime.Value.Subtract(actualStartTime).TotalSeconds;
+                }
+
+                statistics.AverageMessagesPerSecond = durationInSeconds > 0 ? Math.Round((double) statistics.TotalMessagesSent / durationInSeconds, 2) : 0;
             }
```

Comment:
Use Math. ceil instead:

Math.Ceiling( statistics.TotalMessagesSent / durationInSeconds* 100) / 100

Reply:
If we want value upto 2 decimal places, using Math.Round should give the expected value. I checked the value returned in both cases.